### ISSUE TITLE
precious tidy --staged fails if post-checkout hook emits messages on STDOUT

### ISSUE THREAD

**Issue by oalders** [houseabsolute/precious#23]
If you like, I can happily look into fixing this, but I wanted to check with you first. 

Given the following `post-checkout` script:

```bash
cat .git/hooks/post-checkout
#!/bin/bash

echo "foo"
```

If I have a staged file, then `git stash --keep-index` is run by `precious`. This `stash` appears to trigger the `post-checkout` hook.

```
precious tidy --staged
💍 Tidying files staged for a git commit
[precious::precious][ERROR] Failed to run precious: Got unexpected stderr output from `git stash --keep-index`:
foo
```

I *think* the checkout might be happening here, but it's mostly a guess: https://github.com/git/git/blob/master/builtin/stash.c#L1621-L1622

As to why the hook output is sent to `STDERR` rather than `STDOUT`, I think this is likely the explanation: https://github.com/git/git/commit/e258eb4800e30da2adbdb2df8d8d8c19d9b443e4

If I have all of this correct, then maybe the logic which runs `git stash --keep-index` should be checking an exit code rather than `STDERR`?

**Comment by autarch** [comment]
I think this is as simple as changing a `false` to a `true` at `src/basepaths.rs#133`. It would also be nice to have a test for this too.

**Comment by oalders** [comment]
Great! I will add look at this tomorrow.

**Comment by autarch** [comment]
Fixed in v0.1.4.

**Comment by oalders** [comment]
Thanks!!!
